### ISSUE TITLE
kvstreamer: `TestLargeKeys` can fail in Go 1.17.6

### ISSUE THREAD

**Issue by rickystewart** [cockroachdb/cockroach#75180]
```
------- Stdout: -------
=== RUN   TestLargeKeys/vectorize=true#01
    streamer_test.go:168:
          Error Trace:  streamer_test.go:168
                              subtest.go:23
          Error:        Received unexpected error:
                        pq: memory budget exceeded: 6000066 bytes requested, 3000065 currently allocated, 6291456 bytes in budget
          Test:         TestLargeKeys/vectorize=true#01
    --- FAIL: TestLargeKeys/vectorize=true#01 (0.18s)
```

**Comment by yuzefovich** [comment]
Hm, I got no failures after 4 hours of running it on the gceworker with `make stress PKG=./pkg/kv/kvclient/kvstreamer TESTS=TestLargeKeys TESTFLAGS='-test.v -show-logs' STRESSFLAGS='-p 4'`, so closing as not reproducible and will look closer if we see the test flake on master.

**Comment by irfansharif** [comment]
Here's a recent flake from master: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_Test/4170855?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildDeploymentsSection=true&expandBuildTestsSection=true.

**Comment by yuzefovich** [comment]
This time it was
```
202560 panic: streamer-1-unlimited-4: no bytes in account to release, current 1024, free 1026 [recovered]
202561 › panic: streamer-1-unlimited-4: no bytes in account to release, current 1024, free 1026
202562 
```
and I think it's a known flake that will be addressed by #75127.
